doc/GeneratorMigration/Library_Inventory_Guide.md

```diff
@@ -0,0 +1,84 @@
+# Azure SDK for .NET Library Inventory Guide
+
+This document outlines how to generate and maintain the inventory of libraries in the Azure SDK for .NET repository.
+
+## Purpose
+
+The library inventory helps us:
+
+1. Track all libraries in the repository
+2. Categorize them as data plane or management plane
+3. Identify which generator (Swagger or TypeSpec) they use
+4. Plan migrations to newer generators
+
+## Generating the Inventory
+
+To generate the library inventory:
+
+1. Run the `Library_Inventory.py` script from the repository root:
+
+```bash
+python doc/Library_Inventory.py
+```
+
+2. The script will:
+   - Scan all libraries in the `sdk/` directory
+   - Categorize them by type (data plane or management plane)
+   - Identify the generator used (Swagger or TypeSpec)
+   - Generate a markdown report (`doc/Library_Inventory.md`)
+   - Generate a JSON file (`doc/Library_Inventory.json`) for programmatic use
+
+## Inventory Classification Logic
+
+The script uses the following logic to categorize libraries:
+
+### Data Plane vs Management Plane
+
+- **Management Plane**: Libraries that contain "Azure.ResourceManager" or ".Management." in their path
+- **Data Plane**: All other libraries
+
+### Swagger vs TypeSpec (TSP)
+
+The script looks for evidence of TypeSpec usage in the following order:
+
+1. Direct TypeSpec indicators:
+   - `tsp-location.yaml` file with an `emitterPackageJsonPath` value
+   - `src/tspconfig.yaml` file
+   - `src/tsp` directory
+   - `.tsp` files in the src directory
+
+2. TypeSpec references in `src/autorest.md`:
```

Comment:
I don't think we need these checks - the tsp-location.yaml/autorest.md should be sufficient.